sensor/debugger/k8s/fake.go

```diff
@@ -436,6 +437,123 @@ func (f *FakeEventsManager) waitOnMode(events []string) error {
+		reflect.ValueOf(metav1.ListOptions{}),
+	})
+
+	if len(returnVals) != 2 {
+		return fmt.Errorf("expected 2 values from %s. Received: %d", "List", len(returnVals))
+	}
+	errInt := returnVals[len(returnVals)-1].Interface()
+	if errInt != nil {
+		return errInt.(error)
+	}
+
+	errorList := errorhelpers.NewErrorList("for each resource")
+	items := reflect.Indirect(returnVals[0]).FieldByName("Items")
+	if !items.IsValid() {
+		errorList.AddError(errors.New("invalid kind: missing `Items` property"))
+	} else {
+		for i := 0; i < items.Len(); i++ {
+			resource := items.Index(i)
+			if err := execFunc(resource); err != nil {
+				errorList.AddError(err)
+			}
+		}
+	}
+	return errorList.ToError()
+}
+
+// getNameFromObjectMeta returns the name of the resource
+func getNameFromObjectMeta(obj reflect.Value) (string, error) {
+	objMeta := obj.FieldByName("ObjectMeta")
+	if !objMeta.IsValid() {
+		return "", errors.New("resource does not have the ObjectMeta field")
+	}
+	objMetaPtr := objMeta.Addr()
+	getNameFn := objMetaPtr.MethodByName("GetName")
+	if !getNameFn.IsValid() {
+		return "", errors.New("resource does not have the method GetName")
+	}
+	returnVals := getNameFn.Call([]reflect.Value{})
+	if len(returnVals) != 1 {
+		return "", errors.New("call to GetName should return one parameter")
+	}
+	return returnVals[0].String(), nil
+}
+
+// DeleteAllResources deletes all resources in the cluster
+func (f *FakeEventsManager) DeleteAllResources() error {
+	errorList := errorhelpers.NewErrorList("delete all resources")
+	namespaceClientFunc, ok := f.clientMap[namespaceKind]
+	if !ok {
+		errorList.AddStringf("kind %s not found", namespaceKind)
+	}
+	cl := reflect.ValueOf(namespaceClientFunc(""))
+	var namespaces []string
+	if err := f.forEachResource(cl, func(resource reflect.Value) error {
+		name, err := getNameFromObjectMeta(resource)
+		if err != nil {
+			return err
+		}
+		namespaces = append(namespaces, name)
+		return nil
+	}); err != nil {
+		errorList.AddError(err)
+	}
+	resourcesList := []string{
+		secretKind,
+		serviceAccountsKind,
+		roleKind,
+		clusterRoleKind,
+		roleBindingKind,
+		clusterRoleBindingKind,
+		networkPolicyKind,
+		serviceKind,
+		jobKind,
+		podKind,
+		replicaSetKind,
+		replicationControllerKind,
+		daemonSetKind,
+		deploymentKind,
+		statefulSetKind,
+		cronJobKind,
+		namespaceKind,
+		nodeKind,
+	}
+	for _, kind := range resourcesList {
+		clFunc, ok := f.clientMap[kind]
+		if !ok {
+			errorList.AddStringf("kind %s not found", kind)
+			continue
+		}
+		for _, ns := range namespaces {
+			cl = reflect.ValueOf(clFunc(ns))
+			if err := f.forEachResource(cl, func(resource reflect.Value) error {
+				name, err := getNameFromObjectMeta(resource)
+				if err != nil {
+					return err
+				}
+				retVals := runOp(removeAction, cl, reflect.ValueOf(name))
+				if len(retVals) == 0 {
+					return fmt.Errorf("expected 1 or 2 values from %s. Received: %d", removeAction, len(retVals))
+				}
```

Comment:
I guess `len(retVals)` cannot be 3 or more? If it can, then maybe we should change it to

```suggestion
				if len(retVals) < 1 || len(retVals) > 2 {
					return fmt.Errorf("expected 1 or 2 values from %s. Received: %d", removeAction, len(retVals))
				}
```

Reply:
Done!